transmart-copy/src/main/groovy/org/transmartproject/copy/table/Observations.groovy

```diff
@@ -230,28 +215,105 @@ class Observations {
                 }
                 if (batch.size() > 0) {
                     batchCount++
-                    insert.executeBatch(batch.toArray() as Map[])
+                    if (config.partition) {
+                        insertRowsToChildTables(batch, header)
+                    } else {
+                        insert.executeBatch(batch.toArray() as Map[])
+                    }
                 }
                 progressBar.stop()
                 log.info "${batchCount} batches of ${config.batchSize} inserted."
                 return
             }
         }
 
-        if (config.temporaryTable) {
-            // transfer data from temporary table to observations table
-            def columnSpec = columns.collect { it.key }.join(', ')
-            def command = "insert into ${table} (${columnSpec}) select ${columnSpec} from ${temporaryTable}"
-            database.jdbcTemplate.execute(command)
-            tx.flush()
-
-            database.jdbcTemplate.execute("drop table ${temporaryTable}")
-        }
-        database.commit(tx)
         if (config.write) {
             tsvWriter.close()
+        } else {
+            if (config.partition) {
+                createChildTableIndexes()
+                if (config.unlogged) {
+                    setLoggedModeForAllChildTables()
+                }
+            } else if (config.unlogged) {
+                setLoggedMode(table, true)
+            }
         }
+        database.commit(tx)
         log.info "Done loading observations data."
     }
 
+    private void insertRowsToChildTables(List<Map> rows, Map<String, Class> header) {
```

Comment:
Please use LinkedHashMap here, to guarantee that the insert order is preserved in iterations.